File: cmd/lm-desktop/frontend/src/lib/events.ts
import { EventsOn } from '../../wailsjs/runtime/runtime';

export function onBufferChanged(callback: (data: Record<string, any>) => void) {
  return EventsOn('buffer:changed', callback);
}

export function onFileOpened(callback: (data: Record<string, any>) => void) {
  return EventsOn('file:opened', callback);
}

export function onFileSaved(callback: (data: Record<string, any>) => void) {
  return EventsOn('file:saved', callback);
}

export function onCursorMoved(callback: (data: Record<string, any>) => void) {
  return EventsOn('cursor:moved', callback);
}

export function onGraphUpdated(callback: (data: Record<string, any>) => void) {
  return EventsOn('graph:updated', callback);
}

File: internal/wailsplugin/events.go
package wailsplugin

import "github.com/EME130/lazymd/internal/pluginapi"

var eventNameMap = map[pluginapi.EventType]string{
	pluginapi.EventBufferChanged: "buffer:changed",
	pluginapi.EventFileOpened:    "file:opened",
	pluginapi.EventFileSaved:     "file:saved",
	pluginapi.EventFileClosed:    "file:closed",
	pluginapi.EventCursorMoved:   "cursor:moved",
	pluginapi.EventModeChanged:   "mode:changed",
	pluginapi.EventGraphUpdated:  "graph:updated",
}

func WailsEventName(t pluginapi.EventType) string {
	if name, ok := eventNameMap[t]; ok {
		return name
	}
	return string(t)
}
